client-src/elements/chromedash-metadata.ts

```diff
@@ -1,39 +1,42 @@
 import {LitElement, css, html} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
 import {ifDefined} from 'lit/directives/if-defined.js';
 import {SHARED_STYLES} from '../css/shared-css.js';
+import {Channels} from '../js-src/cs-client.js';
 
-class ChromedashMetadata extends LitElement {
-  static get properties() {
-    return {
-      implStatuses: {type: Array}, // Read in chromedash-featurelist.
-      status: {attribute: false}, // Read in chromedash-featurelist.
-      selected: {attribute: false}, // Directly edited in /templates/features.html
-      _className: {attribute: false},
-      _fetchError: {attribute: false},
-      _channels: {attribute: false},
-      _versions: {attribute: false},
-    };
-  }
+interface ImplStatus {
+  key: number;
+  val: string;
+}
 
-  constructor() {
-    super();
-    this.implStatuses = [];
-    this.status = {
-      NO_ACTIVE_DEV: 1,
-      PROPOSED: 2,
-      IN_DEVELOPMENT: 3,
-      BEHIND_A_FLAG: 4,
-      ENABLED_BY_DEFAULT: 5,
-      DEPRECATED: 6,
-      REMOVED: 7,
-      ORIGINTRIAL: 8,
-      INTERVENTION: 9,
-      ON_HOLD: 10,
-      NO_LONGER_PURSUING: 1000,
-    };
-    this._channels = {};
-    this._versions = [];
-  }
+@customElement('chromedash-metadata')
+class ChromedashMetadata extends LitElement {
+  @property({type: Array})
+  implStatuses: ImplStatus[] = [];
+  @property({type: Object, attribute: false})
+  status = {
+    NO_ACTIVE_DEV: 1,
+    PROPOSED: 2,
+    IN_DEVELOPMENT: 3,
+    BEHIND_A_FLAG: 4,
+    ENABLED_BY_DEFAULT: 5,
+    DEPRECATED: 6,
+    REMOVED: 7,
+    ORIGINTRIAL: 8,
+    INTERVENTION: 9,
+    ON_HOLD: 10,
+    NO_LONGER_PURSUING: 1000,
+  };
+  @property({type: String, attribute: false})
+  selected!: string;
+  @state()
+  _className: 'canaryisdev' | 'betaisdev' | undefined;
+  @state()
+  _fetchError!: boolean;
+  @state()
+  _channels: Channels = {};
```

Comment:
I think the right type here is 
```suggestion
  _channels: {
    canary?: number,
    dev?: number,
    beta?: number,
    stable?: number,
  } = {};
```

It also looks like it's never re-assigned, so it may as well be a plain member and not `@state()`, since it'll never cause a re-render.

Reply:
should we initialize them with 0s as they are used in calculating the number of interactions below

Reply:
I would leave the default `undefined` to avoid changing the code's behavior. It'll put some `undefined` values into `_versions`, which I haven't traced through to see what they'll do, but unless you do trace it through, I don't think it's a good idea to change it.

Reply:
`    this._versions = [
      this.implStatuses[this.status.PROPOSED - 1].val,
      this.implStatuses[this.status.IN_DEVELOPMENT - 1].val,
      this.implStatuses[this.status.DEPRECATED - 1].val,
      this._channels.canary.version,
      this._channels.dev.version,
      this._channels.beta.version,
      this._channels.stable.version,
    ];` and `    for (let i = this._channels.canary.version - 1; i >= 1; i--) ` if we just leave it undefined it won't pass the test or even just start it as there're a lot of errors of `Object is possibly 'undefined'.`

Reply:
@DanielRyanSmith might know the code paths better, but I didn't see a reason to think that 0 would have better behavior than undefined. Omitting the value might, but you'd need to trace where the values go to be sure of that.

You can add undefined to the type of _versions, and put `if (this._channels.canary !== undefined)` around the for loop. That's the current behavior since undefined-1 is NaN, which isn't >=1.